ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/StateRef.java

```diff
@@ -67,12 +68,12 @@ public int getIndex() {
      * @return Returns a {@link StateRef} parsed from the specified {@link String} value.
      * @throws IllegalArgumentException if the specified value cannot be parsed.
      */
-    public static StateRef parse(@NotNull final String value) {
+    public static StateRef parse(@NotNull final String value, DigestService digestService) {
```

Comment:
Maybe the `DigestService` should be injected instead?
This looks a bit odd.

And maybe this should be an interface instead too so we don't need implementation detail in the API?
Probably a question for @blsemo or @lankydan ?

Reply:
`StateRef` is supposed to be a lightweight data class that any user can create/parse whenever they need one - I don't want that class itself to be hidden behind an interface/factory pattern.
I don't think we can inject the `DigestService` into `StateRef` as it stands right now, `@CordaInject` doesn't work in random classes, and as it is a concrete class, constructor injection would just shift the problem to the public constructor.
If we don't want to pass in the `DigestService` in this function, then we need to make it an interface, have a factory interface, get a factory off the `UtxoLedgerService` and serve up/parse `StateRef` instances via the factory - which seems like massive overkill for a small data class, and as people have pointed out on slack, would be bad user experience.

Reply:
I think this probably is best we can do right now